### PR TITLE
feat: multiple ports, multiple deployments, override pdb and hpa

### PR DESCRIPTION
**Pull request by AlexanderBrevig**
```yaml
deployment:
  enabled: false
cron:
  enabled: true
  schedule: '30 04 * ? *'

containers:
  - image: ${artifact.metadata.image}
```

New API: 

```yaml
app: mytest
shortname: mytst
team: team

env: prd

ingress:
  host: mytest.dev.entur.io
  trafficType: api

service:
  ports:
    - protocol: TCP
      port: 80
      targetPort: 8080

pdb:
  minAvailable: 25%

hpa:
  spec:
    metrics:
      - type: Resource
        resource:
          name: cpu
          target:
            type: Utilization
            averageUtilization: 80
    behavior:
      scaleDown:
        stabilizationWindowSeconds: 300
        policies:
          - type: Pods
            value: 1
            periodSeconds: 15
      scaleUp:
        stabilizationWindowSeconds: 60
        policies:
          - type: Pods
            value: 2
            periodSeconds: 15

deployment:
  replicas: 1
  labels:
    hello: yes
  prometheus:
    enabled: true
    path: /promhey
    port: 999

postgres:
  enabled: true

#container:
containers:
  - name: ngin1
    image: nginxinc/nginx-unprivileged:latest
    grpc: true
    cpu: 1.1
    cpuLimit: 2.0
    memory: 512
    memoryLimit: 1024
    probes:
      liveness:
        path: "/actuator/health/liveness"
      readiness:
        path: "/actuator/health/readiness"
      startup:
        failureThreshold: 300
    ports:
      - containerPort: 8080
        name: http
        protocol: TCP
      - containerPort: 8778
        name: jolokia
        protocol: TCP
  - name: ngin2
    image: nginxinc/nginx:latest
    cpu: 2.0
    memory: 512
    env:
      - name: hey
        value: yo
    envFrom:
      - configMapRef:
          name: yolo
    probes:
      liveness:
        path: "/actuator/health/liveness"
      readiness:
        path: "/actuator/health/readiness"
      startup:
        failureThreshold: 300
```

### CODE REVIEW

**Comment by assadriaz** [comment]
Ingress template does not support multiple ingress and ingress name not a variable (what if ingress name is different than app name).

**Comment by AlexanderBrevig** [comment]
> 

Added `ingresses`

```yaml
ingresses:
  - trafficType: public
    host: good.io
  - trafficType: api
    rules:
      - host: bad.io
        http:
          paths:
            - backend:
                service:
                  name: something
                  port:
                    number: 666
              pathType: ImplementationSpecific

```

File: charts/common/templates/hpa.yaml [hunk 1]
```diff
@@ -4,28 +4,38 @@
 {{- if (and (not .Values.container.forceReplicas) (or (eq "prd" .Values.env) .Values.container.maxReplicas)) }}
   {{- /* Defaults */}}
   {{- $maxReplicas := .Values.container.maxReplicas | default 10 }}
-  {{- $cpuUtilization := 80 }}
+  {{- $cpuUtilization := 100 }}
```
[the remark this answers is not in the document]
Comment (AlexanderBrevig) [reply charts/common/templates/hpa.yaml:7]: Yep :)

File: charts/common/templates/deployment.yaml [hunk 2]
```diff
@@ -3,9 +3,20 @@
...
+{{- $postgres := .Values.postgres -}}
+{{- $internalPort := .Values.service.internalPort -}}
+{{- $grpc := .Values.grpc -}}
+{{- $labels := .Values.deployment.labels | default .Values.container.labels }}
+{{- $volumes := .Values.deployment.volumes | default .Values.container.volumes }}
+{{- $enabled := .Values.deployment.enabled | default .Values.container.enabled }}
+{{- $prometheus := .Values.deployment.prometheus | default .Values.container.prometheus }}
+{{- $replicas := .Values.deployment.replicas | default .Values.container.replicas }}
+{{- $forceReplicas := .Values.deployment.forceReplicas | default .Values.container.forceReplicas }}
+{{- $terminationGracePeriodSeconds := .Values.deployment.terminationGracePeriodSeconds | default .Values.container.terminationGracePeriodSeconds }}
```
Comment (kagkarlsson) [inline charts/common/templates/deployment.yaml:18]: Funker dette 👍 👏

**Review by sindrel — APPROVED** [review]

**Review by kagkarlsson — APPROVED** [review]
Looking good! Probably a good idea to do a before-and-after diff of an app that has been created using the previous version of the chart.
Also, is there any validation for either deployment or cron?

**Comment by AlexanderBrevig** [comment]
closed #39

**Pull request merged**